plugins/common/tls/config.go

```diff
@@ -206,23 +210,62 @@ func (c *ServerConfig) TLSConfig() (*tls.Config, error) {
 func makeCertPool(certFiles []string) (*x509.CertPool, error) {
 	pool := x509.NewCertPool()
 	for _, certFile := range certFiles {
-		pem, err := os.ReadFile(certFile)
+		cert, err := os.ReadFile(certFile)
 		if err != nil {
 			return nil, fmt.Errorf("could not read certificate %q: %w", certFile, err)
 		}
-		if !pool.AppendCertsFromPEM(pem) {
+		if !pool.AppendCertsFromPEM(cert) {
 			return nil, fmt.Errorf("could not parse any PEM certificates %q: %w", certFile, err)
 		}
 	}
 	return pool, nil
 }
 
-func loadCertificate(config *tls.Config, certFile, keyFile string) error {
-	cert, err := tls.LoadX509KeyPair(certFile, keyFile)
+func loadCertificate(config *tls.Config, certFile, keyFile, privateKeyPassphrase string) error {
+	var (
+		cert      tls.Certificate
+		certBytes []byte
+		keyBytes  []byte
+		err       error
+	)
+	certBytes, err = os.ReadFile(certFile)
```

Comment:
I would prefer the short-hand form here
```suggestion
	certBytes, err := os.ReadFile(certFile)
```
and remove the variable declaration above.

Reply:
??? If you do it it is **equivalent** to declaring the var and **directly afterwards** setting the variable. My suggestion is at _function-level scope_ so I don't see how that is different...